### PR TITLE
feat(api): allow end_user in POST /connections

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Changes

Contributes to https://linear.app/nango/issue/NAN-3746/allow-editing-userorg-info-via-the-api

- POST /connections now accepts end_user 

- Fix openapi spec for credentials
- Fix missing zod schema for API_KEY
- Fix tags and document

---

**Enable `end_user` Support in POST /connections and API Contract Enhancements**

This PR introduces support for passing an optional `end_user` object to the POST /connections endpoint, allowing API consumers to create or update end user data directly alongside connections. The change ensures transactionally safe upserts and links between connections and end users, including support for user tags. Extensive refactoring brings the OpenAPI specification, server-side validation (via shared Zod schemas), and type definitions into alignment-both for new API_KEY and existing credential flows. Existing and new validation and integration tests ensure edge cases and contract adherence.

<details>
<summary><strong>Key Changes</strong></summary>

• ``POST`` /connections now accepts and validates an optional ``end_user`` payload (with id, email, `display_name`, and tags), supporting end user import/upsert and transactional linking to connections.
• `OpenAPI` spec updated-```POST`` /connections` documents the ``end_user`` structure and credential validation (including ``API_KEY`` fix, more precise schemas, improved tags support).
• Backend: Shared, stricter Zod schemas unify and enforce credential and end user input validation across both connection and connect-session flows.
• End user type definitions updated to include `tags`; server logic/process and response formatting expanded to support tags throughout connection lifecycle.
• Test coverage broadened: validation/integration/unit tests extended for ``API_KEY`` cred validation and end user contract edge cases.
• Clean-up and centralization of tag-handling logic: tags limited to 64 key-value string pairs.
• Documentation in code and `OpenAPI` updated for new capabilities and stricter validation.

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• ``API`` handler: packages/server/lib/controllers/connection/`postConnection`.ts
• ``API`` validation: packages/server/lib/helpers/validation.ts
• Connect session: packages/server/lib/controllers/connect/`postSessions`.ts
• Types/interfaces (connection, `endUser`)
• Formatters: packages/server/lib/formatters/`endUser`.ts
• Test suites for connections, connect sessions, end users
• `OpenAPI` documentation: docs-v2/spec.yaml

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/server/lib/controllers/connect/postSessions.ts [hunk 1]
```diff
@@ -48,14 +42,6 @@ export const bodySchema = z
...
             )
             .optional(),
-        tags: z
-            // Please be careful when changing this:
-            // It's a labelling system, if we allow more than string people will store complex data (e.g: nested object) and ask for feature around that
-            // It's an object not a an array of string because customers wants to store layers of origin (e.g: projectId, orgId, etc.)
-            // But they complained a lot about concatenation of string, so object solves that cleanly
-            .record(z.string(), z.string())
-            .refine((v) => Object.keys(v).length < 64, { message: 'Tags can not contain more than 64 keys' })
-            .optional(),
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/connect/postSessions.ts:58]: mistakenly put that at the root, while it should have been in end_user

File: packages/server/lib/controllers/connection/postConnection.ts [hunk 2]
```diff
@@ -263,7 +281,24 @@ export const postPublicConnection = asyncWrapper<PostPublicConnection>(async (re
...
+        await db.knex.transaction(async (trx) => {
+            const endUserRes = await upsertEndUser(trx, { account, environment, endUserPayload: body.end_user });
+            if (endUserRes.isErr()) {
+                res.status(500).send({ error: { code: 'server_error', message: 'Failed to update end user' } });
+                return;
+            }
+
+            endUser = endUserRes.value;
+
+            await linkConnection(trx, { endUserId: endUserRes.value.id, connection: updatedConnection.connection });
```
Comment (TBonnin) [inline packages/server/lib/controllers/connection/postConnection.ts:295]: should the Result be checked and return a 500 if it fails?

File: packages/server/lib/helpers/validation.ts [hunk 3]
```diff
@@ -145,3 +145,18 @@ export const connectionCredentialsGithubAppSchema = z.strictObject({
     app_id: z.string().min(1).max(255),
     installation_id: z.string().min(1).max(255)
 });
+
+export const connectionTagsSchema = z
+    // Please be careful when changing this:
+    // It's a labelling system, if we allow more than string people will store complex data (e.g: nested object) and ask for features around that
+    // + It's an object not a an array of string because customers wants to store layers of origin (e.g: projectId, orgId, etc.)
+    // But they complained a lot about concatenation of string, so an object solves that cleanly
+    .record(z.string(), z.string())
```
Comment (TBonnin) [inline packages/server/lib/helpers/validation.ts:154]: 👍

File: packages/server/lib/helpers/validation.ts [hunk 4]
```diff
@@ -145,3 +145,18 @@ export const connectionCredentialsGithubAppSchema = z.strictObject({
...
+    .refine((v) => Object.keys(v).length < 64, { message: 'Tags can not contain more than 64 keys' });
```
Comment (TBonnin) [inline packages/server/lib/helpers/validation.ts:155]: I am pretty sure some people are gonna want a bigger limit

**Review by TBonnin — APPROVED** [review]

File: packages/server/lib/controllers/connection/postConnection.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/server/lib/controllers/connection/postConnection.ts:295]: this one is throwing

File: packages/server/lib/helpers/validation.ts (hunk 4 above)
Comment (bodinsamuel) [reply packages/server/lib/helpers/validation.ts:155]: maybe, not sure tbh but I prefer putting upper limit and it can be increased
